hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/key/KeyUtils.java

```diff
@@ -51,13 +51,27 @@ public static boolean isEmpty(@Nullable final Key pbjKey) {
             return true;
         }
         if (pbjKey.hasKeyList()) {
-            return !((KeyList) key.value()).hasKeys()
-                    || (((KeyList) key.value()).hasKeys()
-                            && ((KeyList) key.value()).keys().isEmpty());
+            final var keyList = (KeyList) key.value();
+            if (!keyList.hasKeys() || keyList.keys().size() == 0) {
+                return true;
+            }
+            for (final var k : keyList.keys()) {
+                if (!isEmpty(k)) {
+                    return false;
+                }
```

Comment:
Is it possible to have a keylist inside of another keylist? If so, seems like we would want to call recursively.
This method is getting even more complicated

Reply:
It is possible to have `KeyList` under keyList. That is why this `isEmpty` is being called on each key. Do you see something wrong with this ? 
yeah , having isEmpty and isValid for each PBJ generated Key will be better.

Reply:
Oh this is a recursive function, nevermind!